userExtend.ts

```diff
@@ -8,15 +9,26 @@ const setTokens = async () => {
     return
   }
 
-  const token = params.get('__sso')
-  const cache = Vue.prototype.$db.usersCollection
+  rootStore.dispatch('user/clearCurrentUser', {silent: true}).then(() => {
+    // clear cart cache
+    const cartCache = Vue.prototype.$db.cartsCollection
+    cartCache.setItem('current-cart-token', null)
+    cartCache.setItem('current-cart-hash', null)
+    cartCache.setItem('current-cart', null)
 
-  await cache.setItem('current-token', token)
+    const token = params.get('__sso')
+    const cache = Vue.prototype.$db.usersCollection
 
-  if (params.hasOwnProperty('__sst')) {
-    const refreshToken = params.get('__sst')
-    await cache.setItem('current-refresh-token', refreshToken)
-  }
+    cache.setItem('current-token', token)
+
+    // we add to the cache we came from sso, useful for detecting on auth page
+    cache.setItem('source-sso', true)
+
+    if (params.hasOwnProperty('__sst')) {
+      const refreshToken = params.get('__sst')
+      cache.setItem('current-refresh-token', refreshToken)
```

Comment:
@brianmcdo Cache `setItem` returns Promise, are you sure we don't need await here?

Reply:
It doesn't seem to matter much, but updated anyway.